### PR TITLE
feat(test): added react-testing-library to test package

### PR DESCRIPTION
**Pull request by johnpmitsch**
More info on the package: https://github.com/testing-library/react-testing-library and discussion:
https://community.theforeman.org/t/using-react-testing-library-in-katello/18120

## Updated Packages

* [ ] root
* [ ] @theforeman/builder
* [x] @theforeman/test
* [ ] @theforeman/eslint-plugin-foreman
* [ ] @theforeman/stories
* [ ] @theforeman/vendor
* [ ] @theforeman/vendor-dev
* [ ] @theforeman/vendor-core

## PR Type

* [ ] Bugfix
* [x] Feature
* [ ] Code style update (whitespace, formatting, missing semicolons, etc.)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] Other… Please describe:

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

## Does this PR fixes open issues?

* [ ] Yes
* [x] No

## Do you use this PR in another repository?

* [x] Yes
* [ ] No

Used in https://github.com/Katello/katello/pull/8644/files

### CODE REVIEW

**Comment by johnpmitsch** [comment]
@sharvit @ezr-ondrej I updated this PR. We currently have this package in Katello's package.json, is this something we still want in foreman-js? 

If so, should I re-export the package or can it be used normally as-is?

**Comment by ezr-ondrej** [comment]
I'd love to have some integration testing library, so :+1: from me to get this to `foreman-js`.
Not sure what you mean by `re-export` so can't be of help there.

**Comment by johnpmitsch** [comment]
> Not sure what you mean by `re-export` so can't be of help there.

I noticed it here for enzyme

https://github.com/theforeman/foreman-js/blob/2116312d4d9cdabb221a229ae24b9d27db884315/packages/test/index.js#L3-L4

Currently, we use [r-t-lib by importing it directly](https://github.com/Katello/katello/blob/master/webpack/test-utils/react-testing-lib-wrapper.js#L9)

will it continue to function the same coming from the testing package or do changes have to be made?

**Comment by ezr-ondrej** [comment]
ah, right, well I've got no idea, but I guess it's just covering up where the stuff actually comming from (what I don't really like) and I would guess we should be good to import it directly.
Unsure, if there is another reason to it though :thinking:

**Comment by johnpmitsch** [comment]
@sharvit Can you please review this addition?

**Comment by amirfefer** [comment]
what about exporting `react-test-library` via [test/index.js](https://github.com/theforeman/foreman-js/blob/2116312d4d9cdabb221a229ae24b9d27db884315/packages/test/index.js)? 
I'd like to see this in foreman as well :+1:

**Comment by sharvit** [comment]
Thanks @johnpmitsch nice addition.

> will it continue to function the same coming from the testing package or do changes have to be made?

It should act normally and it will allow you to `import { render } from '@theforeman/test';`.
Also, please add documentation regarding when and how to use it, see: https://github.com/theforeman/foreman-js/tree/master/packages/test#usage

**Comment by johnpmitsch** [comment]
@amirfefer @sharvit thanks, I re-exported in test/index.js

I think that we should move away from re-exporting the packages and allow them to be used directly, I added it as a discussion topic at our next meeting :)

For now though, I'll follow the same pattern we have been doing and re-export the library.

File: packages/test/index.js [hunk 1]
```diff
@@ -2,4 +2,5 @@ import MockAdapter from 'axios-mock-adapter';
 
 export * from 'enzyme';
 export * from 'react-redux-test-utils';
+export * from '@testing-library/react';
```
Comment (sharvit) [inline packages/test/index.js:5]: jest-dom shouldn't get exported?

**Comment by johnpmitsch** [comment]
@sharvit @ezr-ondrej are you sure the test packages have to be re-exported?

In Katello, we don't specify `enzyme` in our [dependencies](https://github.com/Katello/katello/blob/master/package.json#L25-L43), but we use it in a lot of places i.e. [here](https://github.com/Katello/katello/blob/63746060ec97dfbfea0908e922a98fbe13d32db8/webpack/components/Search/Search.test.js#L2). Which leads me to believe we are using it from foreman's `@theforeman/test` dependency.

If my logic seems ok, can we test this theory out? I'll remove the re-export line, we merge this and release, then see if the r-t-lib dependencies can be removed from Katello and everything still functions as expected.

**Comment by johnpmitsch** [comment]
I removed the re-export line and added some documentation. I mostly pointed to the official docs, katello's usage as an example, and added some recommendations.

I would like to add a general wrapper for r-t-lib here like we have in Katello but would have to think about how to abstract it to work across foreman and katello. Also, I'm not positive yet if that effort is worth it as our usage of r-t-lib may differ. So it's a bit early for that IMO, but something to keep in mind :smiley:

**Comment by sharvit** [comment]
> I removed the re-export line and added some documentation.

I don't think it would work with the current setup as eslint wouldn’t love the fact you are using a library that is missing from your `package.json`.

We had the same issue with modules coming from `vendor-core` and the solution can be similar:
https://github.com/theforeman/foreman-js/blob/master/packages/eslint-plugin-foreman/lib/src/core.js#L5
https://github.com/theforeman/foreman-js/blob/master/packages/vendor-dev/eslint.extends.js

I am not sure if this is still needed :/
https://github.com/theforeman/foreman-js/blob/master/packages/test/src/babel.config.js#L4

**Comment by johnpmitsch** [comment]
> I don't think it would work with the current setup as eslint wouldn’t love the fact you are using a library that is missing from your `package.json`.

@sharvit yes, we do the same fix in the eslint config in [Katello](https://github.com/Katello/katello/blob/master/webpack/.eslintrc.js#L35-L37). Most dependencies we use are not in our package.json with the recent updates. Ideally, we modify the eslint plugin to better handle this for plugins, but eslint won't be an issue with our configuration right now.

I think we should start using packages directly rather than obfuscating where they methods are coming from, as mentioned earlier in the thread. If plugins aren't set up to do so because of eslint, we need to address that. For now, it is addressed in Foreman and  Katello, and we can talk more about how we might address this in the foreman eslint plugin. Does that sound fair?

**Comment by sharvit** [comment]
> > I don't think it would work with the current setup as eslint wouldn’t love the fact you are using a library that is missing from your `package.json`.
> 
> @sharvit yes, we do the same fix in the eslint config in [Katello](https://github.com/Katello/katello/blob/master/webpack/.eslintrc.js#L35-L37). Most dependencies we use are not in our package.json with the recent updates. Ideally, we modify the eslint plugin to better handle this for plugins, but eslint won't be an issue with our configuration right now.
> 
> I think we should start using packages directly rather than obfuscating where they methods are coming from, as mentioned earlier in the thread. If plugins aren't set up to do so because of eslint, we need to address that. For now, it is addressed in Foreman and Katello, and we can talk more about how we might address this in the foreman eslint plugin. Does that sound fair?

OK, that makes sense. Hopefully, nobody would implement it internally and would update the eslint instead.

**Pull request merged**